### PR TITLE
[FLINK-36258][config] set `HADOOP_CONF_DIR` to config hadoop in YarnFileStageTestS3ITCase

### PR DESCRIPTION
**Pull request by showuon**
## What is the purpose of the change

This PR fixes [FLINK-36258](https://issues.apache.org/jira/browse/FLINK-36258): testRecursiveUploadForYarnS3n failed due to no AWS Credentials provided. In this [PR](https://github.com/apache/flink/pull/25302/files#diff-8614cdbd44bb18064e8b6733b913fa21258487624491e76ac0332a777a77e632L111):
https://github.com/apache/flink/blob/bf393918e5c2a81d4dc49ee82e7019f38f64b517/flink-yarn/src/test/java/org/apache/flink/yarn/YarnFileStageTestS3ITCase.java#L111

we remove the deprecated string configs, and also remove this line to set `HDFS_SITE_CONFIG` in `YarnFileStageTestS3ITCase`, but didn't configure it with environment variable `HADOOP_CONF_DIR`. This PR fixes it by setting it correctly.

## Brief change log

Replace the original `HDFS_SITE_CONFIG` config with the env variable`HADOOP_CONF_DIR`.

## Verifying this change

This change is already covered by existing tests, such as `YarnFileStageTestS3ITCase`.

This change added tests and can be verified as follows:

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: no
  - The S3 file system connector: no

## Documentation

  - Does this pull request introduce a new feature? no
  - If yes, how is the feature documented? not applicable

### CODE REVIEW

**Comment by showuon** [comment]
@XComp @reswqa , this PR should fix the broken `YarnFileStageTestS3ITCase`. Could you help trigger CI to verify it?

File: flink-yarn/src/test/java/org/apache/flink/yarn/YarnFileStageTestS3ITCase.java [hunk 1]
```diff
@@ -78,9 +83,7 @@ static void resetFileSystemConfiguration() {
      * href="https://issues.apache.org/jira/browse/HADOOP-3733">HADOOP-3733</a>).
      */
     private static void setupCustomHadoopConfig(File tempFolder) throws IOException {
-        File hadoopConfig =
-                Files.createTempFile(tempFolder.toPath(), UUID.randomUUID().toString(), "")
-                        .toFile();
+        final File hadoopConfig = new File(tempFolder, "core-site.xml");
```
Comment (showuon) [inline flink-yarn/src/test/java/org/apache/flink/yarn/YarnFileStageTestS3ITCase.java:86]: We need to use `core-site.xml` file name for config discovery.

File: flink-yarn/src/test/java/org/apache/flink/yarn/YarnFileStageTestS3ITCase.java [hunk 2]
```diff
@@ -78,9 +83,7 @@ static void resetFileSystemConfiguration() {
      * href="https://issues.apache.org/jira/browse/HADOOP-3733">HADOOP-3733</a>).
      */
     private static void setupCustomHadoopConfig(File tempFolder) throws IOException {
-        File hadoopConfig =
-                Files.createTempFile(tempFolder.toPath(), UUID.randomUUID().toString(), "")
-                        .toFile();
+        final File hadoopConfig = new File(tempFolder, "hdfs-site.xml");
```
Comment (showuon) [inline flink-yarn/src/test/java/org/apache/flink/yarn/YarnFileStageTestS3ITCase.java:86]: We need to use `hdfs-site.xml` file name for config discovery.

**Review by Sxnan — APPROVED** [review]
@showuon Thanks for the contribution! LGTM.

**Comment by Sxnan** [comment]
@flinkbot run azure

**Pull request merged**